torch_np/_util.py

```diff
@@ -82,3 +82,22 @@ def normalize_axis_tuple(axis, ndim, argname=None, allow_duplicate=False):
         else:
             raise ValueError('repeated axis')
     return axis
+
+
+def expand_shape(arr, axis):
+    # taken from numpy 1.23.x, expand_dims function
+    if type(axis) not in (list, tuple):
+        axis = (axis,)
+    out_ndim = len(axis) + arr.ndim
+    axis = normalize_axis_tuple(axis, out_ndim)
+    shape_it = iter(arr.shape)
+    shape = [1 if ax in axis else next(shape_it) for ax in range(out_ndim)]
+    return shape
+
+def handle_empty_axis(arr, func, *args, **kwds):
+    """Reductions: func(arr, axis=(), *args, **kwds)
+    """
+    shape = expand_shape(arr, axis=0)
+    arr1 = arr.reshape(shape)
+    return func(arr1, axis=0, *args, **kwds)
```

Comment:
I think this is not used anymore. It's now inlined in the wrapper that handles those kwargs.